roles/owncloud/tasks/owncloud.yml

```diff
@@ -43,6 +43,22 @@
   template: src=etc_apache2_sites-available_owncloud.j2 dest=/etc/apache2/sites-available/owncloud.conf group=root
   notify: restart apache
 
+- name: Install ownCloud
+  become: true
+  become_user: www-data
+  command: /var/www/owncloud/occ maintenance:install -n
+    --database=pgsql --database-name="{{ owncloud_db_database }}" --database-user="{{ owncloud_db_username }}" --database-pass="{{ owncloud_db_password }}"
+    --admin-user="{{ main_user_name }}" --admin-pass="{{ owncloud_user_password }}" creates=/var/www/owncloud/config/config.php
+
+- name: Read generated owncloud config
+  command: php -r 'require "/var/www/owncloud/config/config.php"; echo json_encode(array_intersect_key($CONFIG, array_flip(["version", "passwordsalt", "secret"])));'
+  register: owncloud_config_str
+
+- name: Reconfigure owncloud
+  template: src=config.php.j2 dest=/var/www/owncloud/config/config.php group=www-data owner=www-data
+  vars:
+    gen_cfg: "{{ owncloud_config_str.stdout | from_json }}"
```

Comment:
@synchrone I don't think you need this? (`Read generated owncloud config/Reconfigure owncloud`). It causes the [tests to fail](https://travis-ci.org/sovereign/sovereign/builds/214376859):

 ```
$ ansible-playbook --syntax-check -i hosts site.yml
playbook: site.yml
ERROR: vars is not a legal parameter in an Ansible task or handler
The command "ansible-playbook --syntax-check -i hosts site.yml" exited with 1.
```

Here is [how I install and configure owncloud](https://github.com/nodiscc/srv01/tree/master/roles/webapp-owncloud/tasks)

```

[... install packages, system/apache config/modules, ... ]

 - name: copy owncloud main config.php
  template: src=var_www_owncloud_config_config.php.j2 dest=/var/www/owncloud/config/config.php owner=www-data group=www-data

[... set file permissions ...]

- name: finish owncloud installation and create main user
  sudo: yes
  sudo_user: www-data
  command: chdir=/var/www/owncloud/ /usr/bin/php occ  maintenance:install \
           --database="mysql" --database-name="owncloud"  --database-user="owncloud" --database-pass="{{ owncloud_db_pass }}" \
           --admin-user="{{ owncloud_user }}" --admin-pass="{{ owncloud_password }}"

- name: add fqdn to owncloud trusted domains
  sudo: yes
  sudo_user: www-data
  command: chdir=/var/www/owncloud/ /usr/bin/php occ  config:system:set trusted_domains 1 --value={{ srv01_fqdn }}

[ ... install owncloud apps, display completion messages ...]
```

My config.php.j2 is just [this](https://github.com/nodiscc/srv01/blob/master/roles/webapp-owncloud/templates/var_www_owncloud_config_config.php.j2); it is installed _before_ `occ` runs. The `occ` tool just updates it it's own with `passwordsalt/secret...` values if needed.